src/full/decoder.rs

```diff
@@ -138,39 +138,65 @@ impl Decoder {
     /// Returns `Err(RLNCError::InvalidDecodedDataFormat)` if the extracted data
     /// does not follow the expected format (e.g., boundary marker issues).
     pub fn get_decoded_data(self) -> Result<Vec<u8>, RLNCError> {
+        let required_len = self.piece_byte_len * self.required_piece_count;
+        let mut buf = vec![0u8; required_len];
+        let final_len = self.get_decoded_data_into(&mut buf)?;
+
+        buf.truncate(final_len);
+        Ok(buf)
+    }
+
+    /// Recovers the original data and writes it into a pre-allocated buffer.
+    ///
+    /// Returns the final length of the decoded data after removing padding.
+    ///
+    /// Use this over `get_decoded_data` when you want to avoid heap allocation
+    /// and have a pre-allocated buffer ready.
+    pub fn get_decoded_data_into(self, out_buf: &mut [u8]) -> Result<usize, RLNCError> {
```

Comment:
With this API present, we should give user a way to query how large should the buffer be at least. I think for now we can just document that in API doc-comment, it needs to be at least, `self.get_piece_byte_len() * self.get_num_pieces_coded_together()` -bytes.

Reply:
Correct, it is better to keep as it was before.